test/runtests.jl

```diff
@@ -139,6 +141,13 @@ struct IncompleteInterval <: AbstractInterval{Int} end
         @test_throws MethodError duration(1.2..2.4)
     end
 
+    @testset "Unitful interval" begin
+        @test 1.5u"m" in 1u"m" .. 2u"m"
+        @test 1500u"μm" in 1u"mm" .. 1u"m"
+        @test !(500u"μm" in 1u"mm" .. 1u"m")
+        @test 1u"m" .. 2u"m" == 1000u"mm" .. 2000u"mm"
```

Comment:
```suggestion
        @test 1u"m" .. 2u"m" == 1000u"mm" .. 2000u"mm"
        @test_throws ErrorException 1m .. 2s
```
Would it be worth verifying that dimension mismatch throws an error since that was the original issue in #86?
(Note that I have not actually tested the line of code suggested here - I'm not that good with git...)
Other than possibly this, the rest LGTM.

Reply:
Thanks for the suggestion! I think the test should be located around `@test_throws ErrorException 1 .. missing`.
I have added the test `@test_throws ErrorException 1u"m" .. 2u"s"`.